bundles/org.openhab.binding.enphase/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,228 @@
+				<description>Period between updates. The default is 5 minutes, the refresh frequency of the Envoy itself</description>
+				<default>5</default>
+				<advanced>true</advanced>
+			</parameter>
+		</config-description>
+	</bridge-type>
+
+
+	<thing-type id="inverter">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="envoy"/>
+		</supported-bridge-type-refs>
+
+		<label>Inverter</label>
+		<description>Inverter</description>
+
+		<channels>
+			<channel id="lastReportWatts" typeId="last-report-watts"/>
+			<channel id="maxReportWatts" typeId="max-report-watts"/>
+			<channel id="lastReportDate" typeId="last-report-date"/>
+			<channel id="status" typeId="status"/>
+			<channel id="producing" typeId="producing"/>
+			<channel id="communicating" typeId="communicating"/>
+			<channel id="provisioned" typeId="provisioned"/>
+			<channel id="operating" typeId="operating"/>
+		</channels>
+
+		<properties>
+			<property name="partNumber"/>
+		</properties>
+
+		<representation-property>serialnumber</representation-property>
+
+		<config-description>
+			<parameter name="serialnumber" type="text" pattern="[0-9]{12}" required="true">
+				<label>Serial Number</label>
+				<description>The serial number of the inverter</description>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<thing-type id="relay">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="envoy"/>
+		</supported-bridge-type-refs>
+
+		<label>Relay Controller</label>
+		<description>Network system relay controller</description>
+
+		<channels>
+			<channel id="relay" typeId="relay"/>
+			<channel id="line1Connected" typeId="line-connected">
+				<label>Line 1 Connection Status</label>
+			</channel>
+			<channel id="line2Connected" typeId="line-connected">
+				<label>Line 2 Connection Status</label>
+			</channel>
+			<channel id="line3Connected" typeId="line-connected">
+				<label>Line 3 Connection Status</label>
+			</channel>
+			<channel id="status" typeId="status"/>
+			<channel id="producing" typeId="producing"/>
+			<channel id="communicating" typeId="communicating"/>
+			<channel id="provisioned" typeId="provisioned"/>
+			<channel id="operating" typeId="operating"/>
+		</channels>
+
+		<properties>
+			<property name="partNumber"/>
+		</properties>
+
+		<representation-property>serialnumber</representation-property>
+
+		<config-description>
+			<parameter name="serialnumber" type="text" pattern="[0-9]{12}" required="true">
+				<label>Serial Number</label>
+				<description>The serial number of the inverter</description>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<!-- Envoy gateway channels -->
+	<channel-group-type id="envoy-data">
+		<label>Envoy Data</label>
+		<channels>
+			<channel id="wattHoursToday" typeId="watt-hours-today"/>
+			<channel id="wattHoursSevenDays" typeId="watt-hours-seven-days"/>
+			<channel id="wattHoursLifetime" typeId="watt-hours-lifetime"/>
+			<channel id="wattsNow" typeId="watts-now"/>
```

Comment:
Usually around here we measure in the unit `kilowatt hours` kWh so do these channels display in kiloWatt hours or are they really in just watt hours? Should they be called the following and then be left up to the UOM to supply the unit that is shown?
`wattsToday`
`wattsWeek`
`wattsLifetime`
`wattsNow`

Not knowing the details of the API and very little experience implementing UOM, I may be wrong with my thinking on this...

Reply:
No they are watt hours. Because it are UOM channels the user can configure the items to display in kWh.